app/web/src/newhotness/ComponentContextMenu.vue

```diff
@@ -17,13 +17,85 @@ import {
   DropdownMenu,
   DropdownMenuItemObjectDef,
 } from "@si/vue-lib/design-system";
+import { useQuery } from "@tanstack/vue-query";
 import { computed, nextTick, ref } from "vue";
+import { RouteLocationRaw } from "vue-router";
+import { bifrost, useMakeArgs, useMakeKey } from "@/store/realtime/heimdall";
 import { ComponentId } from "@/api/sdf/dal/component";
+import {
+  BifrostActionViewList,
+  ActionPrototypeViewList,
+  BifrostComponent,
+  EntityKind,
+} from "@/workers/types/entity_kind_types";
+import { ActionId, ActionPrototypeId } from "@/api/sdf/dal/action";
 import EraseModal from "./EraseModal.vue";
-// import { useApi } from "./api_composables";
+import { useApi, routes } from "./api_composables";
 
 const contextMenuRef = ref<InstanceType<typeof DropdownMenu>>();
 
+const key = useMakeKey();
+const args = useMakeArgs();
+
+// ================================================================================================
+// This is the location of objects needed to populate menu items.
+const ids = ref<ComponentId[]>([]);
+// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+const id = computed(() => (ids.value.length === 1 ? ids.value[0]! : ""));
```

Comment:
I called these `ids` and `id` to prevent collision with the parameter `componentIds` in the `open` function. We can change this in post if it is not desireable.